Добавление Dagger

Phantom2097 opened the pull request:
Задание 13

Phantom2097 committed:
Добавлен Dagger

akashaev commented on app/src/main/java/ru/phantom/library/di/UseCasesModule.kt:
```diff
@@ -0,0 +1,89 @@
+    fun provideAddItemInLibraryUseCase(
+        repository: ItemsRepository<BasicLibraryElement>
+    ) : AddItemInLibraryUseCase {
+        return AddItemInLibraryUseCase(repository)
```
По-хорошему, тут лучше возвращать реализацию класса, типа так:
AddItemInLibraryUseCase - сделать интерфейсом, который и будет лежать в domain. И рядом добавить его реализацию AddItemInLibraryUseCaseImpl(repository)

@Provides
     fun provideAddItemInLibraryUseCase(
         repository: ItemsRepository<BasicLibraryElement>
     ) : AddItemInLibraryUseCase {
         return AddItemInLibraryUseCaseImpl(repository)
     }


Такой подход позволит разделить описание и реализацию

То же самое справедливо и для @Provides ниже
Прошу прощения, что при предыдущем ревью об этом не сказал

Phantom2097 committed:
Поправил UseCases, теперь они представлены интерфейсами и имеют реализации.
Немного поправил очистку элементов при запросе к api.
Поправил зависимости в корневом файле уровня проекта.

Phantom2097 committed:
fix list fast scroll
fix drawable

Phantom2097 committed:
smallUpdate
